OneCommunityGlobal/HGNRest
# Johny fix bug on create folder for tasks with subtasks

#Description
Fixe bug ( PRIORITY MEDIUM) - 1
Creating sub-tasks in a WBS doesn’t create a new folder and makes the tasks uneditable
Logged in as Admin → Other Links → Projects → AAA USE THIS PROJECT → Click WBS Icon → Choose “Jae's WBS - USE THIS” → Choose task not in folder → Edit → Add Task → Should create folder out of parent and including new task.
## Mainly changes explained:
1 - Was added a new field  called  `childrenQtd`  in the task Schema  `./src/models/task.js` , this field is responsible to count the number of childrens (subtasks)  of a task.
2 - Was created a new function in the ` ./src/controllers/taskController.js` called `updateChildrenQtd` responsible to update the `childrenQtd` when to delete a subtask of a given task.
3 - Then was added a route in the file `./src/routes/taskRouter.js`.

![Captura de Tela (102)](https://user-images.githubusercontent.com/39320057/210471121-4f658011-c6d6-4a1a-b4b5-bc6b15cf219b.png)
![Captura de Tela (103)](https://user-images.githubusercontent.com/39320057/210471129-27c046f7-ff93-4dcc-8248-75dabdac4201.png)
![Captura de Tela (101)](https://user-images.githubusercontent.com/39320057/210471131-12b8aa2f-a8c7-427f-9622-be74a6b699ce.png)

## How to test:
check into current branch
do `npm run build` then
do `npm install` and `...` to run this PR locally


## Note:
The mean functionality is in the front-end please verify the PR related to the frontend and run both.
OBS: "When was creating  subtask for a task  the folder icon was showing up, but when delete all subtasks the folder icon did not goes away".
